Categories in "Accessibility" are incorrect

aditimantri2196 opened the pull request:
## Summary

-The brief provided for "private protected" should be the brief for "protected internal". 
-Removed "private protected" from the category.

Fixes #6206

aditimantri2196 committed:
Categories in "Accessibility" are incorrect

-The brief provided for "private protected" should be the brief for "protected internal" instead. 
-As per my knowledge in c#, "private protected" access modifier does not exist. If it does what does this exactly do?

svick commented on docs/csharp/tour-of-csharp/classes-and-objects.md:
```diff
@@ -58,11 +58,9 @@ Each member of a class has an associated accessibility, which controls the regio
 	- Access limited to the current assembly (.exe, .dll, etc.)
 * `protected internal`
-	- Access limited to the containing class or classes derived from the containing class
+	- Access limited to the containing class or classes derived from the containing type within the same assembly
```
Both the old version and the new version are wrong.

What it should say is that `protected internal` means that access is limited to:

* the containing class
* classes derived from the containing class
* types within the same assembly

BillWagner replied:
The text above is correct. I would write this as:  "Access limited to the containing class, classes derived from the containing class, or classes within the same assembly."

aditimantri2196 replied:
@BillWagner Updated as per your review comment. You can see the required change in latest commit.

BillWagner commented on docs/csharp/tour-of-csharp/classes-and-objects.md:
```diff
@@ -58,11 +58,9 @@ Each member of a class has an associated accessibility, which controls the regio
 * `private`
-	- Access limited to this class
-* `private protected`
-    - Access limited to the containing class or classes derived from the containing type within the same assembly
```
This would be "nice to have", but I'll merge the PR if the text in the `protected internal` is fixed.

aditimantri2196 replied:
@d-dizhevsky Thanks for my knowledge update. It helped.
@Bill Wagner Added private protected back in latest commit, with its original brief.

BillWagner reviewed (changes requested):
Thanks for creating this and fixing it @aditimantri2196 

We apprecia6te it.  There's one change I"d like you to make before I :shipit:.  If you updated line 61, and sync that branch, the PR will be updated. I'll approve the changes and :shipit:.

Just @ - mention me if you have any questions. Thanks again.

aditimantri2196 committed:
Update classes-and-objects.md

-Updated brief for protected internal as per the review comment
-Added private protected back with some modification as per the reference doc.

aditimantri2196 committed:
Update classes-and-objects.md

Added back the private protected with its original brief.

aditimantri2196 commented:
@BillWagner  Made the requested change.

aditimantri2196 committed:
Updated the number of categories of accessiblility

Updated the number of categories of accessiblility from five to six.

BillWagner commented:
Thanks for making the updates @aditimantri2196 I've reviewed the latest version, I'll :shipit: now.

You should see the changes on the live site in the next few days, on our regular publishing cycle.

Thanks again.

BillWagner committed:
quick grammar fix
